plugins/BEdita/API/src/Controller/ConfigController.php

```diff
@@ -0,0 +1,50 @@
+<?php
+/**
+ * BEdita, API-first content management framework
+ * Copyright 2018 ChannelWeb Srl, Chialab Srl
+ *
+ * This file is part of BEdita: you can redistribute it and/or modify
+ * it under the terms of the GNU Lesser General Public License as published
+ * by the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * See LICENSE.LGPL or <http://gnu.org/licenses/lgpl-3.0.html> for more details.
+ */
+
+namespace BEdita\API\Controller;
+
+/**
+ * Controller for `/config` endpoint.
+ *
+ * @since 4.0.0
+ *
+ * @property \BEdita\Core\Model\Table\ConfigTable $Table
+ */
+class ConfigController extends ResourcesController
+{
+    /**
+     * {@inheritDoc}
+     */
+    public $modelClass = 'Config';
+
+    /**
+     * {@inheritDoc}
+     */
+    public function index()
+    {
+        // Add `mine` filter to load common configuration and
```

Comment:
Is this controller only supposed to serve application configuration, and not be used for CRUD operations?

If that's the case, we might avoid extending `ResourcesController` at all:

```php
/**
 * Controller for `/config` endpoint.
 *
 * @since 4.0.0
 *
 * @property \BEdita\Core\Model\Table\ConfigTable $Config
 */
class ConfigController extends AppController
{

    /**
     * {@inheritDoc}
     */
    public $modelClass = 'Config';

    /**
     * Display available configurations.
     *
     * @return void
     */
    public function index()
    {
        $data = $this->Config->find()
            ->find('mine')
            ->find('context', ['core', 'app']);

        $this->set(compact('data'));
    }
}
```

Reply:
I'd avoid creating an `Action`, too. Actions make sense when there's some logic that's worth making reusable. This piece of code is so trivial and so API-specific that IMHO just isn't worth it.